### PR TITLE
DAOS-12334 control: Do not print spurious errors on SPDK setup

### PR DESCRIPTION
**Pull request by tanabarr**
Running DAOS/SPDK via UIO as an unprivileged user is no longer
supported so we don't need to modify the relevant file ownership.

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/control/server/init/setup_spdk.sh [hunk 1]
```diff
@@ -97,11 +97,9 @@ else
...
 		'/sys/class/uio/uio*/device/resource*'; do
 
-		# shellcheck disable=SC2086
-		if list=$(ls -d $glob); then
-			echo -n "RUN: ls -d $glob | xargs -r chown -R "
-			echo "$_TARGET_USER"
-			echo "$list" | xargs -r chown -R "$_TARGET_USER"
+		if compgen -G "${glob}" > /dev/null; then
+			echo "RUN: chown -R ${_TARGET_USER} ${glob}"
+			chown -R "${_TARGET_USER}" "${glob}"
```
Comment (knard-intel) [inline src/control/server/init/setup_spdk.sh:102]: It should not work: the completion will not be done with "..."
```suggestion
			chown -R "${_TARGET_USER}" "$(compgen -G "{glob}")"
```

**Review by knard-intel — CHANGES_REQUESTED** [review]

File: src/control/server/init/setup_spdk.sh (hunk 1 above)
Comment (tanabarr) [reply src/control/server/init/setup_spdk.sh:102]: why should this not be quoted? it will always be a single string

**Comment by tanabarr** [comment]
CI failure identified as known issue DAOS-12631

File: src/control/server/init/setup_spdk.sh (hunk 1 above)
Comment (knard-intel) [reply src/control/server/init/setup_spdk.sh:102]: Quote will prevent globbing. If `glob` variable is equal to `foo*` then `foo_bar` will not be matched.
However, if you want that `glob` to be strictly equal to the file path, then you are right.
I was understanding that you want globbing to take place ?

Comment (tanabarr) [reply src/control/server/init/setup_spdk.sh:102]: that's correct , thank you for spotting this, I've just verified on wolf

Comment (tanabarr) [reply src/control/server/init/setup_spdk.sh:102]: I will also push the fix on master

File: src/control/server/init/setup_spdk.sh [hunk 2]
```diff
@@ -97,11 +97,9 @@ else
...
 		'/sys/class/uio/uio*/device/resource*'; do
 
-		# shellcheck disable=SC2086
-		if list=$(ls -d $glob); then
-			echo -n "RUN: ls -d $glob | xargs -r chown -R "
-			echo "$_TARGET_USER"
-			echo "$list" | xargs -r chown -R "$_TARGET_USER"
+		if compgen -G "${glob}" > /dev/null; then
+			echo "RUN: chown -R ${_TARGET_USER} ${glob}"
+			chown -R "${_TARGET_USER}" ${glob}
```
[the remark this answers is not in the document]
Comment (knard-intel) [reply src/control/server/init/setup_spdk.sh:102]: Ugly workaround to this lint issue
```suggestion
			chown -R "${_TARGET_USER}" "$(compgen -G "${glob}")"
```

**Review by knard-intel — APPROVED** [review]
LGTM

**Review by brianjmurrell — APPROVED** [review]

**Comment by tanabarr** [comment]
CI failure identified as https://daosio.atlassian.net/browse/DAOS-12631

**Pull request merged**